bpfman: Fix panic if XDP prog already loaded (#849)
bpfman/bpfman: docs/getting-started/troubleshooting.md

# Troubleshooting

This section provides a list of common issues and solutions when working with `bpfman`.

## XDP

### XDP Program Fails to Load

When attempting to load an XDP program and the program fails to load:

```console
$ sudo bpfman load image --image-url quay.io/bpfman-bytecode/xdp_pass:latest xdp --iface veth92cd99b --priority 100

> Error: status: Aborted, message: "An error occurred. dispatcher attach: `bpf_link_create` failed", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Tue, 28 Nov 2023 13:37:02 GMT", "content-length": "0"} }

Reviewer: Just a thought we may want to stop adding documentation, or try and minimize the amount that actually has code output from bpfman just because it's becoming a bit of a maintenance nightmare. 

Like in a perfect world we have perfect error variants on `BpfManError` and a troubleshooting guide that goes with that 🤔

Reviewer: Fair enough!

```

The log may look something like this:

```console
Nov 28 08:36:58 ebpf03 bpfman[2081732]: The bytecode image: quay.io/bpfman-bytecode/xdp_pass:latest is signed
Nov 28 08:36:59 ebpf03 bpfman[2081732]: Loading program bytecode from container image: quay.io/bpfman-bytecode/xdp_pass:latest
Nov 28 08:37:01 ebpf03 bpfman[2081732]: The bytecode image: quay.io/bpfman/xdp-dispatcher:v2 is signed
Nov 28 08:37:02 ebpf03 bpfman[2081732]: BPFMAN load error: Error(
                                            "dispatcher attach: `bpf_link_create` failed",
                                        )
```

> The issue may be the there is already an XDP program loaded on the given interface
> that was loaded outside of bpfman.

Reviewer: ```suggestion
The issue may be the there is already an external XDP program loaded on the given interface.
```
